source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,135 +4,382 @@ Sort Results
+Use the ``Sorts.metaTextScore()`` static factory method to build your sort
+criteria to sort by the text score.
+
+.. warning:: Make Sure to Create a Text Index
+
+   You need a :manual:`text index </core/index-text/>` on your collection to
+   perform a text search. See the server manual documentation for more
+   information on how to
+   :manual:`create a text index </core/index-text/#create-text-index>`.
+
+In the following code example, we show how you can use the
+``Sorts.metaTextScore()`` method to sort the results of a text
+search on the :ref:`sort_example collection <sorts-crud-sort-example>`.
+The code example uses the :doc:`Filters </fundamentals/builders/filters>`,
+:doc:`Indexes </fundamentals/builders/indexes>`, and
+:doc:`Projections </fundamentals/builders/projections>` builders.
+The code example performs the following actions:
+
+#. Creates a text index for your
+   :ref:`sort_example collection <sorts-crud-sort-example>`
+   on the ``food`` field. In a production application you would not
+   want to call the ``createIndex()`` method every search, but it is alright for
+   the purpose of this example.
+#. Runs your text search for the phrase "maple donut".
+#. Projects text scores into your query results as the
+   ``score`` field. This projection is optional if your MongoDB instance is
+   running MongoDB 4.4 or later. 
+#. Sorts your results by text score (best match first).
+
+.. code-block:: java
+
+   import com.mongodb.client.model.Sorts;
+   import com.mongodb.client.model.Projections;
+   import com.mongodb.client.model.Filters;
+   import com.mongodb.client.model.Indexes;
+
+   // <MongoCollection setup code here>
+   
+   collection.createIndex(Indexes.text("food"));
+   Bson metaTextScoreSort = Sorts.metaTextScore("score");
+   Bson metaTextScoreProj = Projections.metaTextScore("score");
+   String searchTerm = "maple donut";
+   Bson searchQuery = Filters.text(searchTerm);
+   collection.find(searchQuery)
+            .projection(metaTextScoreProj)
+            .sort(metaTextScoreSort)
+            .into(results);
+   for (Document result : results) {
+         System.out.println(result.toJson());
+   }
+
+The output of the code example above should look something like this: 
+
+.. code-block:: json
+
+   {"_id": 6, "letter": "c", "food": "maple donut", "score": 1.5}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee", "score": 0.75}
+   {"_id": 3, "letter": "a", "food": "maple syrup", "score": 0.75}
+
+.. note:: MongoDB 4.4 or later ``$meta`` Behavior
+
+  When using MongoDB 4.4 or later, projecting ``Projections.metaTextScore()`` 
+  into your ``FindIterable`` instance is not necessary to sort on the text
+  score. In addition, MongoDB 4.4 or later disregards the field name you specify
+  in a ``$meta`` text score aggregation operation used in a sort. This means
+  that the field name argument you pass to ``Sorts.metaTextScore()`` is ignored
+  in MongoDB 4.4 or later.
```

Comment:
Could you find a way to explain this without repeating "MongoDB 4.4 or later" beyond the title and maybe the first sentence? It only seems to distract from the content.

Reply:
Thank you for the note. I am having trouble cutting down "MongoDB 4.4 or later" usage under two times in the note as one time MongoDB 4.4 needs to be an object and the other time it needs to be the subject. One time the reader is using it, the other time the server itself is ignoring some input. I  agree that including "MongoDB 4.4 or later" multiple times is cumbersome, but I am not sure how to get it cleaner than this fix: https://github.com/mongodb/docs-java/pull/66/commits/13a9cd3f88619242918961478e8c9ec0ae5ee063

Reply:
How about introducing the paragraph with the condition and explaining that everything that follows is related to that condition? E.g. "The structure of text searches has changed for MongoDB 4.4 and later. When projecting ... In addition, the field name you specify in a $meta text score aggregation operation used in the sort is ignored..."